### PR TITLE
Optimize Github Action for Compose tests

### PR DESCRIPTION
**Pull request by zion830**
I solved the issue #226 
I used [android emulator runner](https://github.com/marketplace/actions/android-emulator-runner).
When I personally tested, it took about 20 minutes. 
I didn't expect that testing with an emulator takes so long time like this. 💦
스크린샷 2022-09-06 오후 11 30 07

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -14,18 +14,46 @@ on:
...
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            ui:
+              - '**/ui/**'
```
Comment (zion830) [inline .github/workflows/android.yml:27]: I also applied [paths-filter](https://github.com/dorny/paths-filter) to run androidTest conditionally.
The current setting in `on:` runs androidTest whenever a push or pull request occurs.
I added filter to check the location of updated files, the following tasks are started only if the path contains `ui`

* When updated files aren't in the ui package.
스크린샷 2022-09-07 오전 12 01 20

**Review by yeonns — APPROVED** [review]
@zion830 you're right. Running androidTest with emulator takes so many times 😢 
But, it would help a lot to test compose component! 👍
Thanks for updating github action for the latest version of Euphony!

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -14,18 +14,46 @@ on:
...
+          java-version: '11'
+
+      - name: Run Android Tests
+        if: steps.changes.outputs.ui == 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 30
+          script: ./gradlew euphony:connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.package=co.euphony.ui
+          arch: x86_64
+          target: google_apis
```
Comment (designe) [inline .github/workflows/android.yml:43]: here is important part of PR :)
Through `ui condition` check, it is better than basic usage of `reactivecircus/android-emulator-runner`!

**Review by designe — APPROVED** [review]
Hi @zion830 
I really appreciate you made this well-done action script.
It'll be helpful with ui compose development :)

Let me approve & merge with `master` :)

**Pull request merged**